app/models/goal.py

```diff
@@ -1,5 +1,27 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import Integer, String
 from ..db import db
 
 class Goal(db.Model):
-    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    __tablename__ = "goals"
+
+    id: Mapped[int]    = mapped_column(Integer, primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(String, nullable=False)
+
+    tasks: Mapped[list["Task"]] = relationship(
+        "Task",
+        back_populates="goal",
+        cascade="all, delete-orphan"
+    )
+
+    def to_dict(self) -> dict:
+        return {
+            "id":    self.id,
```

Comment:
Nitpick: fix up whitespaces here. Please review the [PEP8 style guide here](https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements) on how to use whitespaces in your code if this topic feels unclear. 

```suggestion
            "id": self.id,
```